### PR TITLE
[test/SSAT] Check flatbuf and protobuf decoder sub-plugins

### PR DESCRIPTION
**Pull request by anyj0527**
- Check existence of  flatbuf and protobuf decoder sub-plugins for some SSAT testcases.

Signed-off-by: Yongjoo Ahn <yongjoo1.ahn@samsung.com>

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: tests/nnstreamer_decoder/runTest.sh [hunk 1]
```diff
@@ -58,6 +58,64 @@ callCompareTest testcase3.con.log testcase3.dec.log 3 "Compare for case 3" 0 0
...
+# Check flatbuf and protobuf libraries are built
+# The tests below are valid only with flatbuf and protobuf sub-plugin
+if [[ -d $PATH_TO_PLUGIN ]]; then
+    ini_path="${PATH_TO_PLUGIN}/ext/nnstreamer/tensor_decoder"
+    if [[ -d ${ini_path} ]]; then
+        check=$(ls ${ini_path} | grep flatbuf.${SO_EXT})
+        if [[ ! $check ]]; then
+            echo "Cannot find flatbuf"
+            report
+            exit
```
Comment (myungjoo) [inline tests/nnstreamer_decoder/runTest.sh:70]: Make a SSAT test case that gives WARNING if FlatBuf does not exist.
(non-critical FAIL)
Then, skip the rest by calling report and exit.

**Comment by anyj0527** [comment]
@myungjoo 
I updated the test after your suggestion. PTAL

File: tests/nnstreamer_decoder/runTest.sh [hunk 2]
```diff
@@ -58,23 +58,47 @@ callCompareTest testcase3.con.log testcase3.dec.log 3 "Compare for case 3" 0 0
...
+# Check existence of tensor_decoder flatbuf sub-plugin
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=3 ! \
+    videoconvert ! video/x-raw,width=64,height=32,framerate=5/1,format=RGB ! \
+    tensor_converter ! tensor_decoder mode=flatbuf ! fakesink" 5 1 0 0
+
+if  (( ${_fail} > 0 ))
+then
+    echo "Cannot find tensor_decoder flatbuf sub-plugin"
+    report
+    exit
```
Comment (myungjoo) [inline tests/nnstreamer_decoder/runTest.sh:70]: If flatbuf does not exist, you need to skip flabuf tests only.
You are skipping protobuf tests, too.

File: tests/nnstreamer_decoder/runTest.sh [hunk 3]
```diff
@@ -58,23 +58,47 @@ callCompareTest testcase3.con.log testcase3.dec.log 3 "Compare for case 3" 0 0
...
+gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=3 ! \
+    videoconvert ! video/x-raw,width=64,height=32,framerate=5/1,format=RGB ! \
+    tensor_converter ! tensor_decoder mode=protobuf ! fakesink" 7 1 0 0
+
+if  (( ${_fail} > 0 ))
+then
+    echo "Cannot find tensor_decoder protobuf sub-plugin"
+    report
+    exit
+fi
```
Comment (myungjoo) [inline tests/nnstreamer_decoder/runTest.sh:93]: Same issue with error detection

**Review by myungjoo — CHANGES_REQUESTED** [review]
I'd suggest to use ```gst-inspect``` along with gst-plugin-path option and ```sed``` to search for supported decoders.

**Comment by anyj0527** [comment]
@myungjoo 
Updated commit use `gst-inspect-1.0` and `grep` to check existence of sub-plugins.
PTAL

File: tests/nnstreamer_decoder/runTest.sh [hunk 4]
```diff
@@ -59,22 +59,36 @@ callCompareTest testcase3.con.log testcase3.dec.log 3 "Compare for case 3" 0 0
...
-## After converting the flatbuf to the tensor, it is tested whether the config is updated in runtime.
-gstTest "--gst-plugin-path=${PATH_TO_PLUGIN} videotestsrc num-buffers=3 ! videoconvert ! video/x-raw, width=640, height=480, framerate=5/1,format=RGB ! \
-    tee name=t ! queue ! multifilesink location=\"test_05_raw_%1d.log\"
-    t. ! queue ! tensor_converter ! tensor_decoder mode=flatbuf ! other/flatbuf-tensor ! tensor_converter ! \
-        tensor_decoder mode=direct_video ! multifilesink location=\"test_05_decoded_%1d.log\" sync=true" 5 "Flatbuf dynamic tensor config update" 0 0 $PERFORMANCE
-callCompareTest test_05_raw_0.log test_05_decoded_0.log 5-1 "Compare for case 5-1" 1 0
-callCompareTest test_05_raw_1.log test_05_decoded_1.log 5-2 "Compare for case 5-2" 1 0
-callCompareTest test_05_raw_2.log test_05_decoded_2.log 5-3 "Compare for case 5-3" 1 0
+# Check existence of tensor_decoder flatbuf sub-plugin
+check=$(gst-inspect-1.0 --gst-plugin-path=${PATH_TO_PLUGIN} tensor_decoder | grep flatbuf)
```
Comment (gichan-jang) [inline tests/nnstreamer_decoder/runTest.sh:63]: This is a part of the decoder test, but wouldn't it be better to check the converter sub-plugin together?

Comment (anyj0527) [reply tests/nnstreamer_decoder/runTest.sh]: I updated this commit, please check.

**Review by gichan-jang — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**